pkg/registry/extensions/rest/storage_extensions.go

```diff
@@ -47,6 +47,25 @@ func (p RESTStorageProvider) NewRESTStorage(apiResourceConfigSource serverstorag
 	return apiGroupInfo, true
 }
 
+// Wraps deploymentstore.DeploymentStorage to override RollbackREST.ProducesObject
+type DeploymentStorage struct {
+	Deployment *deploymentstore.REST
+	Status     *deploymentstore.StatusREST
+	Scale      *deploymentstore.ScaleREST
+	Rollback   *RollbackREST
+}
+
+type RollbackREST struct {
+	deploymentstore.RollbackREST
+}
+
+// override RollbackREST.ProducesObject
+func (r *RollbackREST) ProducesObject(verb string) interface{} {
```

Comment:
this wrap of the method `ProducesObject` is needed to separate the two api groups (`apps` and `extensions`)

Reply:
I don't follow this. I think the theory is that we (eventually will) use header negotiation to determine the type returned/consumed by a subresource, but for now, it is just constant. I guess I would have expected to see the change you made here, and I'm not sure if the change in the other files serves a purpose?

Reply:
The apiserver endpoint installer uses the type of `ProducesObject`'s return value to document the routes. Because `apps` and `extensions` shared the same `RollbackREST`, if we don't wrap the method, the documentation would be "`extensions/rollback` responses with `apps/deploymentstatus`"